src/pages/ProfilePage.tsx

```diff
@@ -267,28 +245,25 @@ function ProfilePage(props) {
     );
 }
 
-ProfilePage.propTypes = propTypes;
-ProfilePage.defaultProps = defaultProps;
 ProfilePage.displayName = 'ProfilePage';
 
-export default compose(
-    withLocalize,
-    withOnyx({
-        personalDetails: {
-            key: ONYXKEYS.PERSONAL_DETAILS_LIST,
-        },
-        session: {
-            key: ONYXKEYS.SESSION,
-        },
-        report: {
-            key: ({route, session}) => {
-                const accountID = Number(lodashGet(route.params, 'accountID', 0));
-                const reportID = lodashGet(ReportUtils.getChatByParticipants([accountID]), 'reportID', '');
-                if ((session && Number(session.accountID) === accountID) || Session.isAnonymousUser() || !reportID) {
-                    return null;
-                }
-                return `${ONYXKEYS.COLLECTION.REPORT}${reportID}`;
-            },
+export default withOnyx<ProfilePageProps, ProfilePageOnyxProps>({
+    personalDetails: {
+        key: ONYXKEYS.PERSONAL_DETAILS_LIST,
+    },
+    session: {
+        key: ONYXKEYS.SESSION,
+    },
+    report: {
+        key: ({route, session}) => {
+            const accountID = Number(route.params?.accountID ?? 0);
+            const reportID = ReportUtils.getChatByParticipants([accountID])?.reportID ?? '';
+
+            if ((Boolean(session) && Number(session?.accountID) === accountID) || SessionActions.isAnonymousUser() || !reportID) {
+                return `${ONYXKEYS.COLLECTION.REPORT}0`;
```

Comment:
`return `${ONYXKEYS.COLLECTION.REPORT}0`;` will return undefined. Why do we need to update from `return null;`
to 
```
return `${ONYXKEYS.COLLECTION.REPORT}0`;` will return undefined
```

Reply:
The key should be always a key according to our types

The key was updated from `null` to  `string`, the actual result didn't changed 

Before -  `OnyxStorage[null]` - undefined
Now -  `OnyxStorage[`${ONYXKEYS.COLLECTION.REPORT}0`]` - undefined